tests/sensors/test_time_delta.py

```diff
@@ -57,7 +58,8 @@ def setup_method(self):
     @pytest.mark.parametrize(
         "data_interval_end, delta, should_deffer",
         [
-            (REFERENCE_TIME.add(hours=-1), timedelta(hours=1), False),
+            (copy(REFERENCE_TIME).add(hours=-1), timedelta(hours=1), False),
```

Comment:
Mu first attempt was with this:
```suggestion
            (REFERENCE_TIME.add(hours=-1, minutes=-1), timedelta(hours=1), False),
```
But later when it worked locally with -1 hour I removed the -1 minute, but I believe adding it will solve the issue

Reply:
Still -> REFERENCE_TIME.add() modifies the REFERENCE_TIME which is pretty bad - because it might not run if you re-run only the second test

Reply:
Why `add()` would change the value of the original variable? It is not in-place method 🤔
```python
>>> import pendulum
>>> d = pendulum.now()
>>> d
DateTime(2024, 7, 13, 22, 4, 21, 948725, tzinfo=Timezone('Europe/Paris'))
>>> d.add(hours=-1)
DateTime(2024, 7, 13, 21, 4, 21, 948725, tzinfo=Timezone('Europe/Paris'))
>>> d
DateTime(2024, 7, 13, 22, 4, 21, 948725, tzinfo=Timezone('Europe/Paris'))
```

Reply:
Yeah. I read it wrong. Looked at the code as well and you are right.